core/src/main/java/nl/nn/adapterframework/align/Json2Xml.java

```diff
@@ -73,35 +75,17 @@ public Json2Xml(ValidatorHandler validatorHandler, List<XSModel> schemaInformati
 
 	@Override
 	public void startParse(JsonValue node) throws SAXException {
-		if (node instanceof JsonObject) {
+		if (node instanceof JsonObject && StringUtils.isEmpty(getRootElement())) {
 			JsonObject root = (JsonObject)node;
-			if (StringUtils.isEmpty(getRootElement())) {
-				if (root.isEmpty()) {
-					throw new SAXException("Cannot determine XML root element, neither from attribute rootElement, nor from JSON node");
-				}
-				if (root.size()>1) {
-					String namesList=null;
-					int i=0;
-					for (String name:root.keySet()) {
-						if (namesList==null) {
-							namesList=name;
-						} else {
-							namesList+=","+name;
-						}
-						if (i++>5) {
-							namesList+=", ...";
-							break;
-						}
-					}
-					throw new SAXException("Cannot determine XML root element, too many names ["+namesList+"] in JSON");
-				}
-				setRootElement((String)root.keySet().toArray()[0]);
-			}
+			List<String> potentialRootElements = new ArrayList<>(root.keySet());
+			potentialRootElements.removeIf(e-> {return e.startsWith(attributePrefix) || e.startsWith(mixedContentLabel);});
+			determineRootElement(potentialRootElements);
+
 			// determine somewhat heuristically whether the json contains a 'root' node:
 			// if the outermost JsonObject contains only one key, that has the name of the root element,
 			// then we'll assume that that is the root element...
-			if (root.size()==1 && getRootElement().equals(root.keySet().toArray()[0])) {
-				node=root.get(getRootElement());
+			if (potentialRootElements.size()==1 && getRootElement().equals(potentialRootElements.get(0))) {
+				node = root.get(getRootElement());
```

Comment:
I think that if this condition is false, then you won't reach this line b/c the above method would have thrown already?